apps/backend/src/app/api/v1/auth/passkey/sign-in/verification-code-handler.tsx

```diff
@@ -0,0 +1,149 @@
+import { createAuthTokens } from "@/lib/tokens";
+import { prismaClient } from "@/prisma-client";
+import { createVerificationCodeHandler } from "@/route-handlers/verification-code-handler";
+import { VerificationCodeType } from "@prisma/client";
+import { verifyAuthenticationResponse } from "@simplewebauthn/server";
+import {decodeClientDataJSON} from "@simplewebauthn/server/helpers";
+import { KnownErrors } from "@stackframe/stack-shared";
+import { signInResponseSchema, yupMixed, yupNumber, yupObject, yupString } from "@stackframe/stack-shared/dist/schema-fields";
+import { StackAssertionError } from "@stackframe/stack-shared/dist/utils/errors";
+import { createMfaRequiredError } from "../../mfa/sign-in/verification-code-handler";
+import { AuthenticationResponseJSON } from "@stackframe/stack-shared/dist/utils/passkey";
+
+export const passkeySignInVerificationCodeHandler = createVerificationCodeHandler({
+  metadata: {
+    post: {
+      summary: "Sign in with a passkey",
+      description: "Sign in with a passkey",
+      tags: ["Passkey"],
+      hidden: true,
+    }
+  },
+  type: VerificationCodeType.PASSKEY_AUTHENTICATION_CHALLENGE,
+  requestBody: yupObject({
+    authenticationResponse: yupMixed<AuthenticationResponseJSON>().required(),
+    code: yupString().required(),
+  }),
+  data: yupObject({
+    challenge: yupString().required()
+  }),
+  method: yupObject({}),
+  response: yupObject({
+    statusCode: yupNumber().oneOf([200]).required(),
+    bodyType: yupString().oneOf(["json"]).required(),
+    body: signInResponseSchema.required(),
+  }),
+  async send() {},
+  async handler(project, _, {challenge}, {authenticationResponse}) {
+
+    if (!project.config.passkey_enabled) {
+      throw new KnownErrors.PasskeyAuthenticationNotEnabled();
+    }
+
+
+    const credentialId = authenticationResponse.id;
+
+
+    // Get passkey from DB with userHandle
+    const passkey = await prismaClient.passkeyAuthMethod.findFirst({
+      where: {
+        credentialId,
+        projectId: project.id,
+      },
+      include: {
+        projectUser: true,
+      },
+    });
+
+
+    if (!passkey) {
+      throw new StackAssertionError("Passkey not found", { credentialId });
```

Comment:
Couldn't this happen on a legitimate user request? We should probably just return a 400 error, right?